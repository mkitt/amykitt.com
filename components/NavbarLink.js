// @flow
import React, { type Node } from 'react'
import css, { hover } from '../styles/css'

type Props = {
  children: Node,
  href: string,
  isActive: boolean,
}

const style = css(
  {
    display: 'inline-block',
    verticalAlign: 'middle',
    fontSize: 12,
    fontWeight: 'bold',
    textDecoration: 'none',
    textTransform: 'uppercase',
    borderBottom: '1px solid transparent',
    transition: 'color 0.2s',
  },
  hover({ color: '#000' }),
)

const activeStyle = css(
  { ...style },
  { borderBottomColor: 'currentColor' },
)

export default ({ children, href, isActive, ...props }: Props) => (
  <a
    className={`${style} ${isActive ? activeStyle : ''}`}
    href={href}
    {...props}
  >
    {children}
  </a>
)
