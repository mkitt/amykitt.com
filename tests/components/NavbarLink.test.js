import React from 'react'
import renderer from 'react-test-renderer'
import NavbarLink from '../../components/NavbarLink'

const noop = () => {}

describe('NavbarLink', () => {
  let component

  beforeEach(() => {
    component = (
      <NavbarLink
        href="/projects"
        isActive
        onMouseover={noop}
      >
        NavbarLink
      </NavbarLink>
    )
  })

  it('shows the correct snapshot for the NavbarLink component tree', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
