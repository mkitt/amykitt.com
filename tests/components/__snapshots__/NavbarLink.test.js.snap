// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarLink shows the correct snapshot for the NavbarLink component tree 1`] = `
<a
  className="css-e3p90y css-863xba"
  href="/projects"
  onMouseover={[Function]}
>
  NavbarLink
</a>
`;
